ocs_ci/deployment/helpers/hypershift_base.py

```diff
@@ -0,0 +1,450 @@
+        cmd = (
+            f"oc get --namespace clusters nodepools | awk '$1==\"{name}\" {{print $3}}'"
+        )
+        return exec_cmd(cmd, shell=True).stdout.decode("utf-8").strip()
+
+    def wait_hosted_cluster_completed(self, name, timeout=3600):
+        """
+        Wait for HyperShift hosted cluster creation to complete
+        :param name: name of the cluster
+        :param timeout: timeout in seconds
+        :return: True if cluster creation completed, False otherwise
+        """
+        logger.info(f"Verifying HyperShift hosted cluster {name} creation is Completed")
+        for sample in TimeoutSampler(
+            timeout=timeout,
+            sleep=60,
+            func=self.get_hosted_cluster_progress,
+            name=name,
+        ):
+            if sample == "Completed":
+                return True
+
+    def wait_for_worker_nodes_to_be_ready(self, name, timeout=2400):
+        """
+        Wait for worker nodes to be ready for HyperShift hosted cluster
+        :param name: name of the cluster
+        :param timeout: timeout in seconds
+        :return: True if worker nodes are ready, False otherwise
+        """
+
+        logger.info(
+            f"Verifying worker nodes to be ready for HyperShift hosted cluster {name}. "
+            f"Max wait time: {timeout} sec "
+        )
+        for sample in TimeoutSampler(
+            timeout=timeout,
+            sleep=60,
+            func=self.worker_nodes_deployed,
+            name=name,
+        ):
+            if sample:
+                return True
+
+    def get_hosted_cluster_kubeconfig_name(self, name):
+        """
+        Get HyperShift hosted cluster kubeconfig, for example 'hcp414-bm2-a-admin-kubeconfig'
+        :param name: name of the cluster
+        :return: hosted cluster kubeconfig name
+        """
+        logger.info(f"Getting kubeconfig for HyperShift hosted cluster {name}")
+        cmd = f"oc get --namespace clusters hostedclusters | awk '$1==\"{name}\" {{print $3}}'"
+        return exec_cmd(cmd, shell=True).stdout.decode("utf-8").strip()
+
+    def download_hosted_clusters_kubeconfig_files(self):
+        """
+        Get HyperShift hosted cluster kubeconfig for multiple clusters
+        :return: list of hosted cluster kubeconfig paths
+        """
+        names = get_hosted_cluster_names()
+
+        kubeconfig_paths = []
+        for name in names:
+            kubeconfig_paths.append(self.download_hosted_cluster_kubeconfig(name))
+
+        return kubeconfig_paths
+
+    def download_hosted_cluster_kubeconfig(self, name: str, auth_path: str = None):
+        """
+        Download HyperShift hosted cluster kubeconfig
+        :param name: name of the cluster
+        :param auth_path: path to download kubeconfig
+        :return: path to the downloaded kubeconfig, None if failed
+        """
+        if not auth_path:
+            auth_path = constants.AUTH_PATH_PATTERN.format(name)
```

Comment:
It is more complicated I think, so I wanted to avoid misunderstanding. 
For instance by the name in PATH we are picking the correct kubeconfig and performing "oc" commands to specific cluster. I wanted to make sure we have a standard in the path name. Other than that I don't see why user(qe) may want to have a custom path for the auth files.
We use "~/clusters/CLUSTER_NAME/openshift-cluster-dir/auth" as a standard here.

Reply:
The reason, why I would make it configurable is, that the `~/clusters/CLUSTER_NAME` convention is specific for our CI environment - jenkins jobs, but it might not be suitable for local execution and definitely it will not be suitable for other teams, if they would like to run our tests - so I would still vote for the option to make it configurable, while the default value could be the current path.

The reason why I previously suggested to place it to the _log_dir_ is, that this path is by default stored on the Shared storage on jenkins job, wile the `~/clusters/...` path has to be explicitly mounted for the particular agent (otherwise it will be stored only locally on the particular temp jenkins agent and will be lost whenever the jenkins agent will be destroyed or broken) - but we will anyway have to ensure that the cluster path will be stored on the Shared storage on proper location and mount the `~/clusters/...` directories properly, so the _log_dir_ is probably not good idea.

So to summarize it, I'm ok with the default value which you have implemented there, but I would propose to add option to change it from configuration file - if necessary.

Reply:
I've added `ENV_DATA.cluster_paths` dependency to store credentials and maybe other docs related to specific cluster. 
The cluster_paths should have the name of the cluster within the path, for example `~/clusters/CLUSTER_NAME/auth` and name should be one of the desired names from variable ENV_DATA.cluster_names

This is one of the cases when I see it will be correct to abort the deployment so I've added this block into HostedClients constructor:

` def __init__(self):
        HyperShiftBase.__init__(self)
        if not ("cluster_names" in config.default_cluster_ctx.ENV_DATA):
            raise ValueError("No 'cluster_names' set to ENV_DATA")
        if not ("cluster_paths" in config.default_cluster_ctx.ENV_DATA):
            raise ValueError("No 'cluster_paths' set to ENV_DATA")
        if not len(config.default_cluster_ctx.ENV_DATA.get("cluster_names")) != len(
            "cluster_paths"
        ):
            raise ValueError(
                "The number of 'ENV_DATA.cluster_names' and 'ENV_DATA.cluster_paths' should be the same"
            )`